File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    
    
    visited=[0]*(n)
    def bfs(start):
        que=deque([start])
        visited[start]=1
        while que:
            now=que.popleft()
            for i in range(n):
                if visited[i]==0 and computers[now][i]==1:
                    que.append(i)
                    visited[i]=1
                    
    for i in range(n):
        for j in range(n):
            if visited[j]==0 and computers[i][j]==1:
                bfs(j)
                answer+=1
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static int graph[][];
    static int len;
    static boolean visited[];
    public int solution(int n, int[][] computers) {
        int answer = 0;
        len = n;
        graph = new int[n][n];
        visited = new boolean[n];

        for(int i=0; i<len; i++) {
            if(!visited[i]) {
                //System.out.println(i);
                bfs(i, computers);
                answer++;
            }
            
        }
        return answer;
    }
    static void bfs(int x, int[][] computers) {
        
        Queue<Integer> que = new ArrayDeque<>();
        que.offer(x);
        visited[x] = true;
        while(!que.isEmpty()) {
            int temp = que.poll();
            for(int i=0; i<len; i++) {
                if(computers[temp][i] == 1 && !visited[i]) {
                    que.offer(i);
                     //System.out.println(i);
                    visited[i] = true;
                   
                }
            }
        }
    }
}